ts/components/registration/stages/RestoreAccount.tsx

```diff
@@ -0,0 +1,321 @@
+  useOnboardHexGeneratedPubKey,
+  useProgress,
+  useRecoveryPassword,
+  useRecoveryPasswordError,
+} from '../../../state/onboarding/selectors/registration';
+import {
+  registerSingleDevice,
+  registrationDone,
+  signInByLinkingDevice,
+} from '../../../util/accountManager';
+import { setSignInByLinking, setSignWithRecoveryPhrase } from '../../../util/storage';
+import { Flex } from '../../basic/Flex';
+import { SessionButton, SessionButtonColor } from '../../basic/SessionButton';
+import { SpacerLG, SpacerSM } from '../../basic/Text';
+import { SessionIcon } from '../../icon';
+import { SessionInput } from '../../inputs';
+import { SessionProgressBar } from '../../loading';
+import { resetRegistration } from '../RegistrationStages';
+import { OnboardDescription, OnboardHeading } from '../components';
+import { BackButtonWithininContainer } from '../components/BackButton';
+import { useRecoveryProgressEffect } from '../hooks';
+import { displayNameIsValid, sanitizeDisplayNameOrToast } from '../utils';
+import { AccountDetails } from './CreateAccount';
+
+type AccountRestoreDetails = AccountDetails & { dispatch: Dispatch; abortSignal?: AbortSignal };
+
+export async function finishRestore(pubkey: string, displayName: string) {
+  await setSignWithRecoveryPhrase(true);
+  await registrationDone(pubkey, displayName);
+
+  window.log.debug(`WIP: [onboarding] restore account: logging in for ${displayName}`);
+  trigger('openInbox');
+}
+
+/**
+ * This will try to sign in with the user recovery password.
+ * If no ConfigurationMessage is received within ONBOARDING_RECOVERY_TIMEOUT, the user will be asked to enter a display name.
+ */
+async function signInAndFetchDisplayName(args: AccountRestoreDetails) {
+  const { recoveryPassword, dispatch, abortSignal } = args;
+
+  try {
+    await resetRegistration();
+    const promiseLink = signInByLinkingDevice(recoveryPassword, 'english', abortSignal);
+    const promiseWait = PromiseUtils.waitForTask(done => {
+      window.Whisper.events.on(
+        'configurationMessageReceived',
+        async (ourPubkey: string, displayName: string) => {
+          window.Whisper.events.off('configurationMessageReceived');
+          await setSignInByLinking(false);
+          dispatch(setHexGeneratedPubKey(ourPubkey));
+          dispatch(setDisplayName(displayName));
+          dispatch(setAccountRestorationStep(AccountRestoration.Finishing));
+          done(displayName);
+        }
+      );
+    }, ONBOARDING_TIMES.RECOVERY_TIMEOUT);
+
+    await Promise.all([promiseLink, promiseWait]);
+  } catch (e) {
+    await resetRegistration();
+    throw e;
+  }
+}
+
+/**
+ * Sign in/restore from seed.
+ * Ask for a display name, as we will drop incoming ConfigurationMessages if any are saved on the swarm.
+ * We will handle a ConfigurationMessage
+ */
+async function signInWithNewDisplayName(args: AccountRestoreDetails) {
+  const { displayName, recoveryPassword, dispatch } = args;
```

Comment:
is this a new way of destructuring?
What happened to the 
```suggestion
async function signInWithNewDisplayName({ displayName, recoveryPassword, dispatch } : AccountRestoreDetails) {

```
syntax? We don't like it anymore?